src/validation.cpp

```diff
@@ -5158,15 +5158,8 @@ void ChainstateManager::LoadExternalBlockFile(
 
                 // Activate the genesis block so normal node progress can continue
                 if (hash == params.GetConsensus().hashGenesisBlock) {
-                    bool genesis_activation_failure = false;
-                    for (auto c : GetAll()) {
```

Comment:
In commit "validation: Don't loop over all chainstates in LoadExternalBlock" (dff5f81433fa8ce1d292a2f6051204a0d7824d6e)

It would be helpful if commit description could clarify whether this is a change in behavior. The "no second chainstate can exist anyway because -reindex would have deleted any snapshot chainstate earlier" part suggests this doesn't change current behavior. But I think in the case when `-loadblock` is being used, not `-reindex`, a snapshot chainstate might actually be loaded and not calling ActivateBestChain would be a (good?) change.

Would just be good to clarify what this change does in terms of externally observable behavior.

Reply:
I added a clarification. The only reason to have this call here is to not block init, and in the `-loadblock` / `assumeutxo` case (which, by the way, I'm not sure anyone has ever tested) this is not an issue / ActivateBestChain() will be called for both chainstates at the end of `ImportBlocks()`.